### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by FireInstall**
Updated the dependencies to the newest (1.20.6) version.
Most importantly is OpenInv with a change in its public API, where "retainPlayer" is now marked for removal.

I know the way I aggressively pushed everything to the newest versions probably means loading this version on older Spigot/Paper versions doesn't work, but since there is version indifferent reflection, loading it in the api defined 1.13 could have been broken anyway.

### CODE REVIEW

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 1]
```diff
@@ -51,13 +38,7 @@
...
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
-import java.util.AbstractMap;
-import java.util.Date;
-import java.util.EnumSet;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.UUID;
+import java.util.*;
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: Please avoid wildcard imports. Explicitly importing each used class makes it clear what other classes a class uses (and what gets changed in it). And this change does not add to the functionality of the PR but just adds noise.

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 2]
```diff
@@ -331,7 +312,7 @@ && getConfig().contains("sync-" + syncType.getKey(), true)) {
         }
 
         if (getServer().getMap((short) 0) == null) {
-            getServer().createMap(getServer().getWorlds().get(0));
+            getServer().createMap(getServer().getWorlds().getFirst());
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: The changes to use `getFirst` seem unnecessary to get support for the new version working so just adds noise. (It even makes a potential backporting to older versions harder although I doubt that will ever be important. Keeping a minimal diff though makes working kith a code base and its history a lot easier though)

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 3]
```diff
@@ -108,7 +89,7 @@ public final class SyncInv extends JavaPlugin {
      * Sync data with all servers in a group when a player logs out
      */
     private boolean syncWithGroupOnLogout;
-    
+
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: Lots of empty line changes that seem to pollute the diff, please revert these and only touch parts that are actually necessary to get the functionality working. If anything this should be a separate PR.

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 4]
```diff
@@ -401,14 +382,14 @@ public boolean onCommand(CommandSender sender, Command cmd, String label, String
...
-     * @param key The key of the message (lang.<key>)
+     * @param key          The key of the message (lang.<key>)
      * @param replacements An array of variables to be replaced with certain strings in the format [var,repl,var,repl,...]
      * @return The message string with colorcodes and variables replaced
      */
     public String getLang(String key, String... replacements) {
         String msg = ChatColor.translateAlternateColorCodes('&', getConfig().getString("lang." + key, getName() + ": &cMissing language key &6" + key));
         for (int i = 0; i + 1 < replacements.length; i += 2) {
-            msg = msg.replace("%" + replacements[i] + "%", replacements[i+1]);
+            msg = msg.replace("%" + replacements[i] + "%", replacements[i + 1]);
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: Another change unrelated to the goal of this PR, please keep it this way/use a separate PR to fix formatting issues.

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 5]
```diff
@@ -631,8 +621,9 @@ public void applyData(PlayerData data, Runnable finished) {
                         player.removePotionEffect(effect.getType());
                     }
                 }
-                if (shouldSync(SyncType.HEALTH))
+                if (shouldSync(SyncType.HEALTH)) {
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: Another unnecessary formatting change.

File: src/main/java/de/minebench/syncinv/SyncInv.java [hunk 6]
```diff
@@ -658,7 +649,7 @@ public void applyData(PlayerData data, Runnable finished) {
                                     map.setLocked(mapData.isLocked());
                                     map.setTrackingPosition(mapData.isTrackingPosition());
                                     map.setUnlimitedTracking(mapData.isUnlimitedTracking());
-                                } catch (NoSuchMethodError ignored) {}
+                                    } catch (NoSuchMethodError ignored) {}
```
Comment (Phoenix616) [inline src/main/java/de/minebench/syncinv/SyncInv.java]: Unless this renders weirdly in the app this looks like it breaks the indention? Anyways, it's unrelated to the update, please revert.

File: src/main/resources/plugin.yml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 name: ${project.name}
 version: '${minecraft.plugin.version}'
-api-version: '1.13'
+api-version: '1.20.6'
```
Comment (Phoenix616) [inline src/main/resources/plugin.yml:3]: I don't think patch versions are valid `api-version` values. According to the [Spigot wiki](https://www.spigotmc.org/wiki/plugin-yml/#optional-attributes) only the minor version gets one. Does this really load without errors and fails to load on older versions?

File: README.md [hunk 8]
```diff
@@ -15,7 +16,7 @@ Provided without any (implied) support nor warranty! (See [GPLv3 license](https:
 1. Install redis, Paper, SyncInv (and optionally OpenInv).
 2. Configure the SyncInv [`config.yml`](https://github.com/Minebench/SyncInv/blob/master/src/main/resources/config.yml):
     1. Specify the `server-name` for each server as set in your proxy config
-    2. Setup the values for the pub-sub connection to your redis server in teh `redis` section
+   2. Setup the values for the pub-sub connection to your redis server in the `redis` section
```
Comment (Phoenix616) [inline README.md]: Thanks for fixing the typo but this seems to have changed the indention too?

File: pom.xml [hunk 9]
```diff
@@ -20,47 +20,47 @@
...
-            <url>https://libraries.minecraft.net/</url>
+            <url>https://repo.papermc.io/repository/maven-public/</url>
         </repository>
         <repository>
             <id>minebench-repo</id>
             <url>https://repo.minebench.de/</url>
         </repository>
+        <repository>
+            <id>jitpack.io</id>
+            <url>https://jitpack.io</url>
```
Comment (Phoenix616) [inline pom.xml]: Please do not use JitPack. They cause lots of issues in the build process and can't produce consistent dependencies. In the past I worked around this by deploying OpenInv to my own repo and I would like to keep it that way. (Or alternatively use a version properly deployed to an official repo of course)

File: pom.xml [hunk 10]
```diff
@@ -20,47 +20,47 @@
     <repositories>
         <repository>
             <id>papermc</id>
-            <url>https://papermc.io/repo/repository/maven-public/</url>
-        </repository>
-        <repository>
-            <id>minecraft-repo</id>
-            <url>https://libraries.minecraft.net/</url>
```
Comment (Phoenix616) [inline pom.xml:27]: The mojang repo was necessary for the authlib dependency. Where is this pulled from now?

**Review by Phoenix616 — CHANGES_REQUESTED** [review]
Thank you for your PR updating it. The actual changes to the OpenInv loading logic seems solid as far as I can tell, will have to test it when I find the time (especially around the edge cases that the player retaining was supposed to help with in the past)

However you seem to have done a lot of changes that don't seem related to the functionality and are really only cosmetic. That makes reviewing it pretty hard as well as messes up the git history, I would advice using a minimal diff policy. Please revert those changes, if you want style changes/fixes then please do those in a separate PR.

File: src/main/java/de/minebench/syncinv/SyncInv.java (hunk 1 above)
Comment (FireInstall) [reply src/main/java/de/minebench/syncinv/SyncInv.java]: well, that's a change the IDE made automatically.
But as you wish I can undo it ^^

File: pom.xml (hunk 10 above)
Comment (FireInstall) [reply pom.xml:27]: Papers repo

File: pom.xml (hunk 9 above)
Comment (FireInstall) [reply pom.xml]: Sadly that's the only current way to catch the official openInv versions. If you wish you can update your Version.
I certainly can't.

File: README.md (hunk 8 above)
Comment (FireInstall) [reply README.md]: No idea what happend there

File: src/main/resources/plugin.yml (hunk 7 above)
Comment (FireInstall) [reply src/main/resources/plugin.yml:3]: they wheren't in the past. But as the minor versions got major technical changes, it's now supported.
I did fetch this from the [Paper wiki](https://docs.papermc.io/paper/dev/plugin-yml)

Comment (FireInstall) [reply src/main/resources/plugin.yml:3]: and yes it does

Comment (Phoenix616) [reply src/main/resources/plugin.yml:3]: Interesting but makes sense I guess. Also kinda annoying how the wiki page you linked contradicts itself in that section. lol "This doesn't include the minor version. [...] ]The valid versions are 1.13 - 1.20.6."

File: pom.xml (hunk 9 above)
Comment (Phoenix616) [reply pom.xml]: 4.4.3 (as well as 4.4.4-SNAPSHOT) are now part of my repo: https://repo.minebench.de/com/lishid/openinvapi/4.4.3/ so you should be able to just adjust the version of the current dependency without adding jitpack.

Comment (FireInstall) [reply pom.xml]: The core not the api. But it's fine you did build the plugin itself as well ^^

Comment (Phoenix616) [reply pom.xml]: Well all artifacts that the project builds are in the repo, if you think the API is now enough (it wasn't in the past) then you can use that too I guess.

Comment (FireInstall) [reply pom.xml]: You did misunderstand me.
The API isn't enough. However, you have sent the link for the API not the core ^^

File: src/main/resources/plugin.yml (hunk 7 above)
Comment (FireInstall) [reply src/main/resources/plugin.yml:3]: yeah. My guess is that this is a fairly new change and nobody paid attention to the other appearance.

File: pom.xml (hunk 9 above)
Comment (Phoenix616) [reply pom.xml]: Aah, right, makes sense 😅

**Review by Phoenix616 — APPROVED** [review]
Looks good now. Thanks again! I will do some testing myself before merging though just to be sure it works in the different cases (don't want to risk people blindly updating to an untested version of such an essential plugin)
